src/map/entities/battleentity.cpp

```diff
@@ -695,6 +695,8 @@ uint16 CBattleEntity::CHR()
 uint16 CBattleEntity::ATT(uint16 slot)
 {
     TracyZoneScoped;
+    XI_DEBUG_BREAK_IF(slot >= sizeof(m_Weapons) / sizeof(CItemEquipment*));
```

Comment:
If CItemEquipment* is a nullptr (size 0), is this going to create a problem?

Reply:
Shouldn't be? I tested this originally with 1H weapons and saw no issue.

Reply:
This here is just testing for sizeof(CItemEQuipment*). That will always be the size of the class. 
sizeof(m_Weapons) / sizeof(CItemEquipment*) returns the length of the m_Weapons array

Reply:
@paladindamarus FYI minimum size of a class is 4 bytes even if empty:
https://www.cprogramming.com/tutorial/size_of_class_object.html